src/modules/project-config.ts

```diff
@@ -169,6 +168,55 @@ export class ProjectConfig {
         return maybeComponentConfig ? maybeComponentConfig : new ComponentConfig(componentId);
     }
 
+    /**
+     * @param onlyInstalled only retrieves the installed packages for the project. Defaults to false.
+     * @returns all used packages by the project.
+     */
+    getPackages(onlyInstalled: boolean = false): PackageConfig[] {
+        if (onlyInstalled) {
+            return this._packages.filter((pkg) => pkg.isPackageInstalled());
+        }
+
+        return this._packages;
+    }
+
+    getPackageConfig(packageName: string, onlyInstalled: boolean = false): PackageConfig | undefined {
+        return this.getPackages().find((config) => config.getPackageName() === packageName);
+    }
+
+    /**
+     * Adds a new packageConfig to the project. Multiple packages with the same name and version will not be added. However adding
+     * a package which has a different version will update the version.
+     *
+     * @param packageConfig the packageConfig to add.
+     * @returns true if the package was added successfully, false otherwise.
+     */
+    addPackageConfig(packageConfig: PackageConfig): boolean {
+        const existingPackage = this.getPackages().find((config) => config.getPackageName() === packageConfig.getPackageName());
```

Comment:
shouldnt this work with:

const existingPackage = this.getPackageConfig(packageConfig.getPackageName())